### PR TITLE
#53 Selectable

### LINKED ISSUE #53: SelectableList data screen
I have an idea of DataScreen -> SelectableList

It should be for screens like this
image

### PR DESCRIPTION
**Pull request by kopecmi8**

### CODE REVIEW

File: packages/datascreens/src/selectableList.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { IList, ISelectableList } from "./types";
+
+type Constructor<TEntity> = new (...args: any[]) => IList<TEntity>;
+
+export type SelectableListViewModel<TViewModel, TEntity> = TViewModel & ISelectableList<TEntity>;
+
+function Selectable<TEntity, TBase extends Constructor<TEntity>>(Base: TBase) {
+  return decorate(
+    class SelectableList extends Base implements ISelectableList<TEntity> {
+      selectedItems: IObservableArray<TEntity> = observable([]);
```
Comment (gius) [inline packages/datascreens/src/selectableList.ts]: Can we use `Set` here?

**Review by gius — CHANGES_REQUESTED** [review]

File: packages/datascreens/src/selectableList.ts (hunk 1 above)
Comment (kopecmi8) [reply packages/datascreens/src/selectableList.ts]: for sure

File: packages/datascreens/src/selectableList.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          return null;
+        } else {
+          return false;
+        }
+      }
+
+      set allItemsSelected(selectAll: boolean | null) {
+        this.selectedItems.clear();
+        if (selectAll) {
+          this.selectedItems = observable(new Set([...this.items]));
```
Comment (gius) [inline packages/datascreens/src/selectableList.ts:27]: If we recreate the observable set, wouldn't it break existing observers? 
Could we reuse the existing instance of the set?

You might want to check https://gitlab.eman.cz/pudplus/pud-frontend/-/blob/master/src/models/dataList.ts for inspiration.

**Comment by gius** [comment]
Obsolete, solved in v1

**Pull request closed without merging**